src/ol/render/webgl/MixedGeometryBatch.js

```diff
@@ -412,13 +465,44 @@ class MixedGeometryBatch {
     }
   }
 
+  /**
+   * @param {string} featureUid Feature uid
+   * @param {GeometryBatchItem} entry The entry to add
+   * @return {GeometryBatchItem} the added entry
+   * @private
+   */
+  addRefToEntry_(featureUid, entry) {
+    const currentRef = this.uidToRef.get(featureUid);
+    const ref = currentRef || this.freeGlobalRef.pop() || ++this.globalCounter;
```

Comment:
```suggestion
    const ref = currentRef || this.freeGlobalRef.pop() || this.globalCounter++;
```

Otherwise refs are 1-based, isn't that a little counter-intuitive?

Reply:
As discussed, starting from 1 allows to distinguish from the white color. I am adding a comment.